TAGradingServer/account/account-rubric.php

```diff
@@ -558,30 +555,48 @@ function display_files($file, &$output, $part, $indent = 1) {
-                            <td>
-                                <select name="grade-{$question['question_part_number']}-{$question['question_number']}" id="changer" class="grades" style="width: 65px; height: 25px; min-width:0px;" onchange="calculatePercentageTotal();" {$grade_select_extra}>
+                        </tr>
 HTML;
 
-    for ($i = 0; $i <= $question['question_total'] * 2; $i++) {
-        $output .= '<option' . (($i * 0.5) == $question['grade_question_score'] ? " selected" : "") . '>' . round(($i * 0.5), 1) . '</option>';
-    }
-
     $comment = ($question['grade_question_comment'] != "") ? "in" : "";
     $output .= <<<HTML
-                                </select>
-                            </td>
-                            <td><strong>{$question['question_total']}</strong></td>
-                        </tr>
-                        <tr>
-                            <td colspan="3" style="padding:0; border-top:none;">
-                                <div class="accordian-body collapse {$comment}" id="rubric-{$c}">
-                                    <textarea name="comment-{$question["question_part_number"]}-{$question["question_number"]}" rows="2" style="width:100%; padding:0px; resize:none; margin:0px 0px; border-radius:0px; border:none; padding:5px; border-left:3px #DDD solid; float:left; margin-right:-28px;" placeholder="Message for the student..." comment-position="0">{$question['grade_question_comment']}</textarea>
+    <tr style="background-color: #f9f9f9;">
+                            <td style="white-space:nowrap; vertical-align:middle; text-align:center;"><input type="number" id="test-{$question["question_part_number"]}-{$question["question_number"]}" class="grades" name="grade-{$question['question_part_number']}-{$question['question_number']}" value="{$question['grade_question_score']}" min="0" max="{$question['question_total']}" step="0.5" placeholder="&plusmn;0.5" onchange="validateInput_{$question["question_part_number"]}_{$question["question_number"]}(0.5); calculatePercentageTotal();" style="width:50px; resize:none;"></textarea><strong> / {$question['question_total']}</strong></td>
+                            <td style="width:100%; padding:0px">
+                                <div id="rubric-{$c}">
+                                    <textarea name="comment-{$question["question_part_number"]}-{$question["question_number"]}" onkeyup="autoResizeComment(event);" rows="4" style="width:100%; height:100%; padding:0px; resize:none; margin:0px 0px; border-radius:0px; border:none; padding:5px; border-left:3px #DDD solid; float:left; margin-right:-28px;" placeholder="Message for the student..." comment-position="0">{$question['grade_question_comment']}</textarea>
+HTML;
+    $output .= <<<HTML
+    <script>
+    // delta in this function is the incremental step of points, currently hardcoded to 0.5pts
+    function validateInput_{$question["question_part_number"]}_{$question["question_number"]}(delta){
```

Comment:
Move this outside the foreach loop and make the definition be:
```
function validateInput(question_part_number, question_number, question_total, delta) {
```
and then update the definition from where you're calling it as well as the variables in the function.